#
#
# patch "libpurple/buddyicon.c"
#  from [a73df57ec69cebed25d67518c291fe548c233eed]
#    to [d2aa70ec997238e584b40dc3d6058c3788aa6e45]
#
============================================================
--- libpurple/buddyicon.c	a73df57ec69cebed25d67518c291fe548c233eed
+++ libpurple/buddyicon.c	d2aa70ec997238e584b40dc3d6058c3788aa6e45
@@ -1161,7 +1161,8 @@ purple_buddy_icons_init()
 	                                        g_free, NULL);
 	pointer_icon_cache = g_hash_table_new(g_direct_hash, g_direct_equal);
 
-	cache_dir = g_build_filename(purple_user_dir(), "icons", NULL);
+    if (!cache_dir)
+    	cache_dir = g_build_filename(purple_user_dir(), "icons", NULL);
 
 	purple_signal_connect(purple_imgstore_get_handle(), "image-deleting",
 	                      purple_buddy_icons_get_handle(),

